tests/integration_tests/data_steward/retraction/retract_deactivated_pids_test.py

```diff
@@ -0,0 +1,283 @@
+EXPECTED_KEPT_ROWS_QUERY = ('SELECT * '
+                            'FROM {dataset_id}.{table_id} '
+                            'WHERE person_id != {pid} '
+                            'OR(  person_id = {pid} '
+                            'AND {date_column} < (SELECT deactivated_date '
+                            'FROM {dataset_id}.{pid_table_id} '
+                            'WHERE person_id = {pid}))')
+EXPECTED_DROPPED_ROWS_QUERY_END_DATE = (
+    'SELECT * '
+    'FROM {dataset_id}.{table_id} '
+    'WHERE person_id = {pid} '
+    'AND (CASE WHEN {end_date_column} IS NOT NULL THEN {end_date_column} >= '
+    '(SELECT deactivated_date '
+    'FROM `{dataset_id}.{pid_table_id}` '
+    'WHERE person_id = {pid} ) ELSE CASE WHEN {end_date_column} IS NULL THEN '
+    '{start_date_column} >= ( '
+    'SELECT deactivated_date '
+    'FROM `{dataset_id}.{pid_table_id}` '
+    'WHERE person_id = {pid}) END END)')
+EXPECTED_KEPT_ROWS_QUERY_END_DATE = (
+    'SELECT * '
+    'FROM {dataset_id}.{table_id} '
+    'WHERE person_id != {pid} '
+    'OR (person_id = {pid} '
+    'AND (CASE WHEN {end_date_column} IS NOT NULL THEN {end_date_column} < '
+    '(SELECT deactivated_date '
+    'FROM `{dataset_id}.{pid_table_id}` '
+    'WHERE person_id = {pid} ) ELSE CASE WHEN {end_date_column} IS NULL THEN '
+    '{start_date_column} < ( '
+    'SELECT deactivated_date '
+    'FROM `{dataset_id}.{pid_table_id}` '
+    'WHERE person_id = {pid}) END END))')
+INSERT_PID_TABLE = ('INSERT INTO {dataset_id}.{pid_table_id} '
+                    '(person_id, deactivated_date) '
+                    'VALUES{person_research_ids}')
+
+
+class RetractDeactivatedEHRDataBqTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.hpo_id = 'fake'
+        self.project_id = app_identity.get_application_id()
+        if 'test' not in self.project_id:
+            raise RuntimeError(
+                f"Make sure the project_id is set to test.  project_id is {self.project_id}"
+            )
+        self.bq_dataset_id = bq_utils.get_dataset_id()
+        self.bq_sandbox_dataset_id = get_sandbox_dataset_id(self.bq_dataset_id)
+        self.ticket_number = 'DCXXX'
+        self.pid_table_id = 'pid_table'
+        self.pid_table_id_list = [
+            self.project_id + '.' + self.bq_dataset_id + '.' + 'pid_table'
+        ]
+        self.deactivated_ehr_participants = [(1, '2010-01-01'),
+                                             (2, '2010-01-01'),
+                                             (5, '2010-01-01')]
+        self.client = bq.get_client(self.project_id)
+
+    @mock.patch(
+        'retraction.retract_deactivated_pids.get_date_info_for_pids_tables')
+    def test_integration_queries_to_retract_from_fake_dataset(
```

Comment:
This test passes, but should fail due to logic errors.  It passes because the errors are in `get_date_info_for_pids_tables`, which is mocked, and in functions called by `get_date_info_for_pids_tables`.  Instead of mocking the entire function, mock the calls to the external resources only and set their return values as they are expected to look.  See the comment in `retract_deactivated_pids.py` for more info on the error and the comments in the unit tests to understand why this error was logically missed.